core/src/tauri/command.rs

```diff
@@ -66,21 +63,7 @@ pub async fn restart(state: tauri::State<'_, State>) -> Result<(), TauriError> {
     .instrument(span)
     .await;
 
-    if let Ok(path) = current_binary(&Env::default()) {
-        match Command::new(path).spawn() {
-            Ok(_) => {
-                info!("Spawned new process. Waiting to restart.");
-
-                // wait a second
-                tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
-
-                std::process::exit(0);
-            }
-            Err(e) => error!("Error restarting: {}", e),
-        }
-    }
-
-    Ok(())
+    tauri::process::restart(&Env::default());
```

Comment:
switching over to our own restart function lets us give the backend a few seconds to shut down